From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yanbing Zhao <zzzz@mail.ustc.edu.cn>
Date: Mon, 26 Oct 2015 21:10:02 +0800
Subject: v1.6


diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java b/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
index 22a7a12..0fd37e3 100644
--- a/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
@@ -1,6 +1,7 @@
 package com.github.ustc_zzzz.fmltutor.common;
 
 import com.github.ustc_zzzz.fmltutor.block.BlockLoader;
+import com.github.ustc_zzzz.fmltutor.crafting.CraftingLoader;
 import com.github.ustc_zzzz.fmltutor.creativetab.CreativeTabsLoader;
 import com.github.ustc_zzzz.fmltutor.item.ItemLoader;
 
@@ -19,7 +20,7 @@ public class CommonProxy
 
     public void init(FMLInitializationEvent event)
     {
-
+        new CraftingLoader();
     }
 
     public void postInit(FMLPostInitializationEvent event)
diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/crafting/CraftingLoader.java b/src/main/java/com/github/ustc_zzzz/fmltutor/crafting/CraftingLoader.java
new file mode 100644
index 0000000..645d5ed
--- /dev/null
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/crafting/CraftingLoader.java
@@ -0,0 +1,50 @@
+package com.github.ustc_zzzz.fmltutor.crafting;
+
+import com.github.ustc_zzzz.fmltutor.block.BlockLoader;
+import com.github.ustc_zzzz.fmltutor.item.ItemLoader;
+
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.IFuelHandler;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+public class CraftingLoader
+{
+    public CraftingLoader()
+    {
+        registerRecipe();
+        registerSmelting();
+        registerFuel();
+    }
+
+    private static void registerRecipe()
+    {
+        GameRegistry.addShapedRecipe(new ItemStack(ItemLoader.goldenEgg), new Object[]
+        {
+                "###", "#*#", "###", '#', Items.gold_ingot, '*', Items.egg
+        });
+        GameRegistry.addShapedRecipe(new ItemStack(BlockLoader.grassBlock), new Object[]
+        {
+                "##", "##", '#', Blocks.vine
+        });
+        GameRegistry.addShapelessRecipe(new ItemStack(Blocks.vine, 4), BlockLoader.grassBlock);
+    }
+
+    private static void registerSmelting()
+    {
+        GameRegistry.addSmelting(BlockLoader.grassBlock, new ItemStack(Items.coal), 0.5F);
+    }
+
+    private static void registerFuel()
+    {
+        GameRegistry.registerFuelHandler(new IFuelHandler()
+        {
+            @Override
+            public int getBurnTime(ItemStack fuel)
+            {
+                return Items.diamond != fuel.getItem() ? 0 : 12800;
+            }
+        });
+    }
+}
