From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yanbing Zhao <zzzz@mail.ustc.edu.cn>
Date: Wed, 9 Mar 2016 21:43:51 +0800
Subject: v3.1.1


diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java b/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
index 371413f..51f01d9 100644
--- a/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/common/CommonProxy.java
@@ -6,6 +6,7 @@ import com.github.ustc_zzzz.fmltutor.command.CommandLoader;
 import com.github.ustc_zzzz.fmltutor.crafting.CraftingLoader;
 import com.github.ustc_zzzz.fmltutor.creativetab.CreativeTabsLoader;
 import com.github.ustc_zzzz.fmltutor.enchantment.EnchantmentLoader;
+import com.github.ustc_zzzz.fmltutor.entity.EntityLoader;
 import com.github.ustc_zzzz.fmltutor.fluid.FluidLoader;
 import com.github.ustc_zzzz.fmltutor.item.ItemLoader;
 import com.github.ustc_zzzz.fmltutor.potion.PotionLoader;
@@ -27,6 +28,7 @@ public class CommonProxy
         new BlockLoader(event);
         new OreDictionaryLoader(event);
         new PotionLoader(event);
+        new EntityLoader(event);
     }
 
     public void init(FMLInitializationEvent event)
diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/common/EventLoader.java b/src/main/java/com/github/ustc_zzzz/fmltutor/common/EventLoader.java
index e66ab9b..a05fcf7 100644
--- a/src/main/java/com/github/ustc_zzzz/fmltutor/common/EventLoader.java
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/common/EventLoader.java
@@ -4,6 +4,8 @@ import com.github.ustc_zzzz.fmltutor.achievement.AchievementLoader;
 import com.github.ustc_zzzz.fmltutor.block.BlockLoader;
 import com.github.ustc_zzzz.fmltutor.client.KeyLoader;
 import com.github.ustc_zzzz.fmltutor.enchantment.EnchantmentLoader;
+import com.github.ustc_zzzz.fmltutor.entity.EntityGoldenChicken;
+import com.github.ustc_zzzz.fmltutor.item.ItemLoader;
 import com.github.ustc_zzzz.fmltutor.potion.PotionLoader;
 
 import net.minecraft.block.Block;
@@ -11,6 +13,7 @@ import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.Minecraft;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.item.EntityTNTPrimed;
 import net.minecraft.entity.passive.EntityPig;
 import net.minecraft.entity.player.EntityPlayer;
@@ -97,6 +100,16 @@ public class EventLoader
     {
         if (!event.world.isRemote)
         {
+            ItemStack heldItem = event.entityPlayer.getHeldItem();
+            if (ItemLoader.goldenEgg.equals(heldItem.getItem()))
+            {
+                EntityLiving entityLiving = new EntityGoldenChicken(event.world);
+                BlockPos pos = event.pos;
+                entityLiving.setPositionAndUpdate(pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5);
+                --heldItem.stackSize;
+                event.world.spawnEntityInWorld(entityLiving);
+                return;
+            }
             BlockPos pos = event.pos;
             Entity tnt = new EntityTNTPrimed(event.world, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, null);
             event.world.spawnEntityInWorld(tnt);
diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java
new file mode 100644
index 0000000..f3a92e4
--- /dev/null
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java
@@ -0,0 +1,30 @@
+package com.github.ustc_zzzz.fmltutor.entity;
+
+import com.github.ustc_zzzz.fmltutor.item.ItemLoader;
+
+import net.minecraft.entity.passive.EntityChicken;
+import net.minecraft.world.World;
+
+public class EntityGoldenChicken extends EntityChicken
+{
+    public EntityGoldenChicken(World worldIn)
+    {
+        super(worldIn);
+    }
+
+    @Override
+    public void onLivingUpdate()
+    {
+        super.onLivingUpdate();
+    }
+
+    @Override
+    protected void dropFewItems(boolean arg1, int arg2)
+    {
+        if (this.rand.nextInt(10) == 0)
+        {
+            this.dropItem(ItemLoader.goldenEgg, 1);
+        }
+        super.dropFewItems(arg1, arg2);
+    }
+}
diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityLoader.java b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityLoader.java
new file mode 100644
index 0000000..e6041d5
--- /dev/null
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityLoader.java
@@ -0,0 +1,30 @@
+package com.github.ustc_zzzz.fmltutor.entity;
+
+import com.github.ustc_zzzz.fmltutor.FMLTutor;
+
+import net.minecraft.entity.Entity;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.registry.EntityRegistry;
+
+public class EntityLoader
+{
+    private static int nextID = 0;
+
+    public EntityLoader(FMLPreInitializationEvent event)
+    {
+        registerEntity(EntityGoldenChicken.class, "GoldenChicken", 80, 3, true);
+        registerEntityEgg(EntityGoldenChicken.class, 0xffff66, 0x660000);
+    }
+
+    private static void registerEntityEgg(Class<? extends Entity> entityClass, int eggPrimary, int eggSecondary)
+    {
+        EntityRegistry.registerEgg(entityClass, eggPrimary, eggSecondary);
+    }
+
+    private static void registerEntity(Class<? extends Entity> entityClass, String name, int trackingRange,
+            int updateFrequency, boolean sendsVelocityUpdates)
+    {
+        EntityRegistry.registerModEntity(entityClass, name, nextID++, FMLTutor.instance, trackingRange, updateFrequency,
+                sendsVelocityUpdates);
+    }
+}
diff --git a/src/main/resources/assets/fmltutor/lang/en_US.lang b/src/main/resources/assets/fmltutor/lang/en_US.lang
index 7355cd2..3c37748 100644
--- a/src/main/resources/assets/fmltutor/lang/en_US.lang
+++ b/src/main/resources/assets/fmltutor/lang/en_US.lang
@@ -37,3 +37,5 @@ commands.position.usage=/position [player]
 commands.position.success=The position of %1$s is %2$s in world %3$s
 
 fluid.mercury=Mercury
+
+entity.fmltutor.GoldenChicken.name=Golden Chicken
diff --git a/src/main/resources/assets/fmltutor/lang/zh_CN.lang b/src/main/resources/assets/fmltutor/lang/zh_CN.lang
index 667e60a..cc7b077 100644
--- a/src/main/resources/assets/fmltutor/lang/zh_CN.lang
+++ b/src/main/resources/assets/fmltutor/lang/zh_CN.lang
@@ -37,3 +37,5 @@ commands.position.usage=/position [玩家]
 commands.position.success=玩家 %1$s 处于名为 %3$s 的世界，其坐标为 %2$s
 
 fluid.mercury=水银
+
+entity.fmltutor.GoldenChicken.name=黄金鸡
