From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yanbing Zhao <zzzz@mail.ustc.edu.cn>
Date: Wed, 27 Apr 2016 17:31:57 +0800
Subject: v3.1.4


diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/client/entity/model/ModelGoldenChicken.java b/src/main/java/com/github/ustc_zzzz/fmltutor/client/entity/model/ModelGoldenChicken.java
index 04157e7..2983e77 100644
--- a/src/main/java/com/github/ustc_zzzz/fmltutor/client/entity/model/ModelGoldenChicken.java
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/client/entity/model/ModelGoldenChicken.java
@@ -66,8 +66,22 @@ public class ModelGoldenChicken extends ModelBase
 
     protected float getWingSpeed(Entity entity, float rotateFloat)
     {
-        return (float) (((EntityGoldenChicken) entity).getEntityAttribute(EntityGoldenChicken.wingSpeed)
+        float wingSpeed = (float) (((EntityGoldenChicken) entity).getEntityAttribute(EntityGoldenChicken.wingSpeed)
                 .getAttributeValue() * rotateFloat);
+        switch (entity.getDataWatcher().getWatchableObjectByte(16))
+        {
+        case 4:
+            return wingSpeed / 2;
+        case 3:
+            return wingSpeed / 4;
+        case 2:
+            return wingSpeed * 4;
+        case 1:
+            return wingSpeed * 2;
+        case 0:
+        default:
+            return wingSpeed;
+        }
     }
 
     @Override
diff --git a/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java
index 2528922..7a72f58 100644
--- a/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java
+++ b/src/main/java/com/github/ustc_zzzz/fmltutor/entity/EntityGoldenChicken.java
@@ -6,6 +6,8 @@ import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.attributes.IAttribute;
 import net.minecraft.entity.ai.attributes.RangedAttribute;
 import net.minecraft.entity.passive.EntityChicken;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.world.World;
 
 public class EntityGoldenChicken extends EntityChicken
@@ -20,6 +22,38 @@ public class EntityGoldenChicken extends EntityChicken
     }
 
     @Override
+    protected void entityInit()
+    {
+        super.entityInit();
+        this.dataWatcher.addObject(16, new Byte((byte) 0));
+    }
+
+    @Override
+    public void writeEntityToNBT(NBTTagCompound tagCompound)
+    {
+        super.writeEntityToNBT(tagCompound);
+        tagCompound.setByte("WingSpeedMultiplier", this.dataWatcher.getWatchableObjectByte(16));
+    }
+
+    @Override
+    public void readEntityFromNBT(NBTTagCompound tagCompund)
+    {
+        super.readEntityFromNBT(tagCompund);
+        this.dataWatcher.updateObject(16, tagCompund.getByte("WingSpeedMultiplier"));
+    }
+
+    @Override
+    public boolean interact(EntityPlayer player)
+    {
+        if (!super.interact(player))
+        {
+            byte b = this.dataWatcher.getWatchableObjectByte(16);
+            this.dataWatcher.updateObject(16, new Byte((byte) ((b + 1) % 5)));
+        }
+        return true;
+    }
+
+    @Override
     protected void applyEntityAttributes()
     {
         super.applyEntityAttributes();
